### PR TITLE
LPS-159060 Add feature flag to new external reference code field

### PR DESCRIPTION
**Pull request by seltong**

### CODE REVIEW

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -788,6 +790,21 @@ private ObjectDefinition _addObjectDefinition(
...
 			"createDate", false, false);
 
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-158821")) &&
+			!objectDefinition.isSystem()) {
+
+			_objectFieldLocalService.addSystemObjectField(
+				userId, objectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+				ObjectEntryTable.INSTANCE.externalReferenceCode.getName(),
+				dbTableName, ObjectFieldConstants.DB_TYPE_STRING, null, false,
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectDefinitionLocalServiceImpl.java:800]: Hi @seltong,

For this case you can just replace `dbTableName` to `ObjectEntryTable.INSTANCE.getTableName()`

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
@seltong, there is a conflict in this file.

please, could you resolve?

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
LGTM! :tada:

**Comment by seltong** [comment]
ci:forward

**Pull request closed without merging**